src/inputs/internal/SelectFieldBase.tsx

```diff
@@ -178,7 +197,11 @@ export function SelectFieldBase<O, V extends Value>(props: BeamSelectFieldBasePr
   async function maybeInitLoad() {
     if (!Array.isArray(maybeOptions)) {
       setFieldState((prevState) => ({ ...prevState, optionsLoading: true }));
-      const { options } = await maybeOptions.load();
+      const loadedOptions = (await maybeOptions.load()).options;
+      // Ensure the `unset` option is prepended to the top of the list if `unsetLabel` was provided
+      const options = !unsetLabel
+        ? loadedOptions
+        : getOptionsWithUnset(unsetLabel, (await maybeOptions.load()).options);
```

Comment:
Do we need a 2nd `await` here, or just use `loadedOptions`?

Reply:
You're right. This was a copy & paste mistake.